frappe/model/base_document.py

```diff
@@ -928,6 +928,40 @@ def _validate_selects(self):
 					)
 				)
 
+	def _validate_radios(self):
+		if frappe.flags.in_import:
+			return
+
+		for df in self.meta.get_radio_fields():
+			if df.fieldname == "naming_series" or not (self.get(df.fieldname) and df.options):
+				continue
+
+			options = (df.options or "").split("\n")
+
+			# if only empty options
+			if not filter(None, options):
+				continue
+
+			# strip and set
+			self.set(df.fieldname, cstr(self.get(df.fieldname)).strip())
+			value = self.get(df.fieldname)
+
+			if (
+				value not in options
+				and not (frappe.flags.in_test and value.startswith("_T-"))
+				and not df.allow_others
+			):
+				# show an elaborate message
+				prefix = _("Row #{0}:").format(self.idx) if self.get("parentfield") else ""
+				label = _(self.meta.get_label(df.fieldname))
+				comma_options = '", "'.join(_(each) for each in options)
+
+				frappe.throw(
+					_('{0} {1} cannot be "{2}". It should be one of "{3}"').format(
```

Comment:
Don't split one logical message into 2 parts. It makes translations harder. Only values should be substituted.